test/support/utils.rb

```diff
@@ -91,6 +89,22 @@ def create_pseudo_terminal(boot_options: "-r debug/run")
 
     private
 
+    # use this to start a debug session with the test program
+    def start_debugger_with(program)
+      write_temp_file(strip_line_num(program))
+      inject_lib_to_load_path
+      pid = spawn("#{RUBY} -r debug/open #{temp_file_path}")
+      sleep(0.5) # wait for the debug server to start
```

Comment:
it is advanced topic: if you use PTY you can wait for the debugger start.

Reply:
if you don't mind, I'll make this improvement in another PR

Reply:
if you specify the socket file path, it is enough:

```ruby
while !File.exist?(path)
  sleep 0.1
end
```

Reply:
added 👍